src/module_3/model_selection_converted.md

```diff
@@ -0,0 +1,524 @@
+```
+
+
+```python
+print("Train set size:", len(train_data)/(len(train_data)+len(test_data)))
+print("Test set size:", len(test_data)/(len(train_data)+len(test_data)))
+```
+
+Overall, the proportion of examples for each class is maintained well between the train and test sets. Despite the class imbalance, the proportion remains consistent across both sets. This suggests that stratifying the data during the train-test split may not be necessary in this case, as the imbalance is preserved in both sets without intervention. However, it's important to note that this imbalance might affect the performance of some classifiers, and strategies such as oversampling or undersampling of the minority class may be considered to address this issue.
+
+
+```python
+train_class_percentages = train_data['outcome'].value_counts(normalize=True) * 100
+test_class_percentages = test_data['outcome'].value_counts(normalize=True) * 100
+
+print("Percentage of examples for each class in train set:")
+print(train_class_percentages)
+print("\nPercentage of examples for each class in test set:")
+print(test_class_percentages)
+```
+
+The approach described above was a way to split the data without considering the temporal reference. Initially, I assumed that temporal information might not be crucial as there was already data available on the days since the last purchase, etc. However, upon further consideration, I realized the importance of incorporating temporal information into the data splitting process.
+
+To address this, I devised another approach for splitting the data based on this temporal reference. This new approach involves dividing the dataset into training and testing sets while ensuring that the temporal sequence is preserved. By doing so, we aim to account for any temporal dependencies or patterns in the data, which may be critical for accurately modeling and predicting future outcomes.
+
+**NEW SPLITTING**
+
+When separating data, it's crucial to consider the date on which orders are placed for several reasons. Firstly, the temporal aspect provides valuable insights into patterns and trends over time, which can significantly impact the performance of predictive models. By incorporating temporal information into the data splitting process, we ensure that the training data represents past observations, while the testing or validation data represents future observations, mimicking real-world scenarios.
+
+Moreover, not considering the date of orders can lead to a phenomenon known as data leakage. Data leakage occurs when information from the future or unseen data is inadvertently included in the training process, leading to overly optimistic performance estimates of the model. In the context of order data, if future information, such as orders placed after a certain date, is included in the training set, the model may learn patterns that do not generalize well to unseen data, resulting in poor model performance when deployed in production.
```

Comment:
The key here is that you could have in `train` future behaviour that will give you extra information when predicting `test`. That information comes from the future and won't be available in production. Thus, the data leakage.